h2h/production-code/week6/get_data.py

```diff
@@ -4,85 +4,131 @@
 import pandas as pd
 import time
 
-def player_in_game_dataset_encoded(season_start, season_end = 2018, read_filepath = None, write_filepath = None, write_filepath_unencoded = None):
+def make_vals_and_labels(read_encoded = None, read_unencoded = None, write_encoded = None, season_start = 1997, season_end = 2018, write_unencoded = None):
+    """ Makes an encoded dataset into usable numpy arrays.
+
+    Generates or reads a dataset that contains a record of games, with columns
+    representing the players and rows representing games. A player has 1 or -1
+    on a row if they were home or away players in that game, respectively.
+    Then, separates the target column and the data, and stores the feature list,
+    to be used by the user (in theory to train a machine learning model).
+
+    Args:
+        read_encoded: Without this arg, this method will generate a new
+            dataset. If given, it will instead read a CSV with the given name,
+            and then separate into values and labels.
+        read_unencoded: Used only if you want to call
+            _encode_player_in_game_dataset. See downstream description.
+        write_encoded: Used only if you want to call
+            _encode_player_in_game_dataset. See downstream description.
+        season_start: Used only if you want to call
+            _encode_player_in_game_dataset. See downstream description.
+        season_end: Used only if you want to call
+            _encode_player_in_game_dataset. See downstream description.
+        write_unencoded: Used only if you want to call
+            _encode_player_in_game_dataset. See downstream description.
+
+    Returns:
+        - values: numpy array of dimension [number_of_games, number_of_players]
+        - labels: numpy array of dimension [number_of_games, 1]
+        - feature_list: list of feature names, dimension [1, number_of_players]
+
+    Raises:
+        None
+    """
+    if read_encoded: #already have encoded csv... just read
+        dataset = pd.read_csv(read_encoded) #no need to eval, all 0 and 1
+    else:
+        if read_unencoded: #have unencoded csv, just need to encode
+            dataset = _encode_player_in_game_dataset(read_unencoded = read_unencoded)
+        else: #don't have anything, need to generate from scratch
+            dataset = _encode_player_in_game_dataset(season_start = season_start, season_end = season_end, write_encoded = write_encoded, write_unencoded = write_unencoded)
+
+    labels = np.array(dataset['label'])
+    values = ml_record.drop(['label'], axis = 1)
```

Comment:
Maybe I'm crazy, but I can't find ml_record anywhere else

Reply:
good catch, that was the old name... will change